src/components/app/details/triggerView/BranchRegexModal.tsx

```diff
@@ -40,42 +44,109 @@ export default function BranchRegexModal({
                     <Close />
                 </button>
             </div>
         )
     }
 
+    const renderRegexInfo = () => (
+        <div className="dc__border-bottom--b2">
+            <InfoColourBar
+                classname="info_bar dc__no-border-radius dc__no-border-imp"
+                message="Commit will be fetched from the provided branch. This can be changed later."
+                Icon={Info}
+            />
+        </div>
+    )
+
+    const getErrorMessage = (_regexValue) => {
+        if (!_regexValue.value) {
+            return REQUIRED_FIELD_MSG
+        }
+        if (_regexValue.isInvalid) {
+            return BRANCH_REGEX_MODAL_MESSAGING.NoMatchingBranchName
+        }
+        return ''
+    }
+
+    const renderBranchRegexContent = () => (
+        <div className="ci-trigger__branch-regex-wrapper px-20 fs-13 dc__overflow-scroll mxh-400">
+            {material?.map((mat, index) => {
+                const _regexValue = regexValue[mat.gitMaterialId] || {}
+                return (
+                    mat.regex && (
+                        <div className="dc__border-bottom py-20" key={`regex_${mat.id}`}>
+                            <div className="flex left dc__gap-14">
+                                {getGitProviderIcon(mat.gitMaterialUrl, 'icon-dim-24')}
+                                <span>
+                                    <div className="fw-6 fs-14">{mat.gitMaterialName}</div>
+                                    <div className="pb-12">
+                                        {BRANCH_REGEX_MODAL_MESSAGING.MatchingBranchName}&nbsp;
+                                        <span className="fw-6">
+                                            {getBranchRegexName(mat.gitMaterialId) || mat.regex}
+                                        </span>
+                                    </div>
+                                </span>
+                            </div>
+                            <div className="ml-36-imp">
+                                <CustomInput
+                                    name="name"
+                                    data-testid={`branch-name-matching-regex-textbox${index}`}
+                                    tabIndex={index}
+                                    placeholder={BRANCH_REGEX_MODAL_MESSAGING.MatchingBranchNameRegex}
+                                    rootClassName="w-95-imp"
+                                    value={_regexValue.value}
+                                    onChange={(e) => handleRegexInputValue(mat.gitMaterialId, e.target.value, mat)}
+                                    autoFocus
+                                    error={getErrorMessage(_regexValue)}
+                                />
+                            </div>
+                        </div>
+                    )
+                )
+            })}
+        </div>
+    )
     const renderMaterialRegexFooterNextButton = () => {
         let _isDisabled = true
```

Comment:
use reduce

Reply:
I think some would be more efficient as Here reduce() will always iterate over the entire array unless explicitly stopped by returning, which may not be as efficient as methods like some(). What you suggest?